File: BatchTestTool/java/java automated tester/Test/to_be_tested/GomokuGame.java
package to_be_tested;

public class GomokuGame {

    public int boardSize;

    public char[][] board;

    public char currentPlayer;

    public GomokuGame(int boardSize) {
        this.boardSize = boardSize;
        this.board = new char[boardSize][boardSize];
        for (int i = 0; i < boardSize; i++) {
            for (int j = 0; j < boardSize; j++) {
                board[i][j] = ' ';
            }
        }
        this.currentPlayer = 'X';
    }

    public boolean makeMove(int row, int col) {
        if (board[row][col] == ' ') {
            board[row][col] = currentPlayer;
            currentPlayer = (currentPlayer == 'X') ? 'O' : 'X';
            return true;
        }
        return false;
    }

    public char checkWinner() {
        int[][] directions = { { 0, 1 }, { 1, 0 }, { 1, 1 }, { 1, -1 } };
        for (int i = 0; i < boardSize; i++) {
            for (int j = 0; j < boardSize; j++) {
                if (board[i][j] != ' ') {
                    for (int[] direction : directions) {
                        if (checkFiveInARow(i, j, direction)) {
                            return board[i][j];
                        }
                    }
                }
            }
        }
        return ' ';
    }

    public boolean checkFiveInARow(int row, int col, int[] direction) {
        int dx = direction[0];
        int dy = direction[1];
        int count = 1;
        char symbol = board[row][col];
        for (int i = 1; i < 5; i++) {
            int new_row = row + dx * i;
            int new_col = col + dy * i;
            if (new_row < 0 || new_row >= boardSize || new_col < 0 || new_col >= boardSize) {
                return false;
            }
            if (board[new_row][new_col] != symbol) {
                return false;
            }
            count++;
        }
        return count == 5;
    }
}

File: ClassEval_T/cpp/solution/GomokuGame.cpp
#include <vector>
#include <optional>

class GomokuGame {
public:
    explicit GomokuGame(int board_size);
    bool make_move(int row, int col);
    std::optional<char> check_winner() const;
    std::vector<std::vector<char>> getBoard() const;
    bool _check_five_in_a_row(int row, int col, std::pair<int, int> direction) const;
    int board_size;
    std::vector<std::vector<char>> board;
    char current_player;
};

GomokuGame::GomokuGame(int board_size)
    : board_size(board_size),
    board(board_size, std::vector<char>(board_size, ' ')),
    current_player('X')
{}

bool GomokuGame::make_move(int row, int col) {
    if (board[row][col] == ' ') {
        board[row][col] = current_player;
        current_player = (current_player == 'X') ? 'O' : 'X';
        return true;
    }
    return false;
}

std::optional<char> GomokuGame::check_winner() const {
    const std::vector<std::pair<int, int>> directions = { {0, 1}, {1, 0}, {1, 1}, {1, -1} };
    for (int row = 0; row < board_size; ++row) {
        for (int col = 0; col < board_size; ++col) {
            if (board[row][col] != ' ') {
                for (const auto& direction : directions) {
                    if (_check_five_in_a_row(row, col, direction)) {
                        return board[row][col];
                    }
                }
            }
        }
    }
    return std::nullopt;
}

std::vector<std::vector<char>> GomokuGame::getBoard() const {
    return board;
}

bool GomokuGame::_check_five_in_a_row(int row, int col, std::pair<int, int> direction) const {
    int dx = direction.first;
    int dy = direction.second;
    int count = 1;
    char symbol = board[row][col];
    for (int i = 1; i < 5; ++i) {
        int new_row = row + dx * i;
        int new_col = col + dy * i;
        if (!(0 <= new_row && new_row < board_size && 0 <= new_col && new_col < board_size)) {
            return false;
        }
        if (board[new_row][new_col] != symbol) {
            return false;
        }
        ++count;
    }
    return count == 5;
}
